shell/app-shell/elements/sharing/user-context.js

```diff
@@ -156,16 +157,20 @@ customElements.define('user-context', class extends Xen.Debug(Xen.Base, log) {
       }
     }, ['users-stores-keys']));
   }
-  async _updateSystemUser(context, userid, coords, state) {
-    const {user, userStore, userContext} = state;
-    if (userContext) {
-      await userContext.dispose();
-      state.userContext = null;
-    }
-    // if the `userid` has changed before we finished cleaning up, re-validate
-    if (state.userid !== userid) {
-      this._invalidate();
-    } else {
+  async _updateSystemUser(props, state) {
+    if (!state.disposingUser) {
+      const {user, userStore, userContext} = state;
+      if (userContext) {
+        state.disposingUser = true;
+        state.userContext = null;
+        try {
+          await userContext.dispose();
+        } catch (x) {
+          //
```

Comment:
Perhaps log something here?

Reply:
I don't imagine this will ever actually execute, I just put the try block there so the semaphore pattern was complete. I imagine that sounds weird, but that was my thinking.